hack/runtime-migrator/internal/comparator/comparator.go

```diff
@@ -0,0 +1,54 @@
+package comparator
+
+import (
+	"github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	"github.com/kyma-project/infrastructure-manager/hack/shoot-comparator/pkg/shoot"
+)
+
+type Result struct {
+	Equal bool
+	Diff  []Difference
+}
+
+type Difference struct {
+	ShootName  string
+	LeftShoot  v1beta1.Shoot
+	RightShoot v1beta1.Shoot
+	Message    string
+}
+
+func CompareShoots(leftShoot, rightShoot v1beta1.Shoot) (Result, error) {
+
+	differences, err := compare(leftShoot, rightShoot)
+	if err != nil {
+		return Result{}, err
+	}
+
+	equal := len(differences) == 0
+
+	return Result{
+		Equal: equal,
+		Diff:  differences,
+	}, nil
+}
+
+func compare(leftShoot, rightShoot v1beta1.Shoot) ([]Difference, error) {
+	var differences []Difference
+
+	matcher := shoot.NewMatcher(leftShoot)
+	equal, err := matcher.Match(rightShoot)
+	if err != nil {
+		return nil, err
+	}
+
+	if !equal {
+		diff := Difference{
+			ShootName:  leftShoot.Name, // assumption that leftShoot and rightShoot have the same name
```

Comment:
should that assumption be checked? If this is supposed to be always true, maybe it's safer to even crash if it ever isn't?

Reply:
Thx, I will add the step with checking if left == right